app/static/tests/e2e/options.etest.ts

```diff
@@ -0,0 +1,96 @@
+        await expect(await page.inputValue(":nth-match(#model-params input, 4)")).toBe("2");
+    });
+
+    test("can collapse and expand parameters panel", async ({ page }) => {
+        // collapse
+        await page.click(".collapse-title a");
+        await expect(await page.getAttribute(".collapse-title a i", "data-name")).toBe("chevron-down");
+        await expect(await page.locator("#model-params")).toBeHidden();
+
+        // expand
+        await page.click(".collapse-title a");
+        await expect(await page.getAttribute(".collapse-title a i", "data-name")).toBe("chevron-up");
+        await expect(await page.locator("#model-params")).not.toBeHidden();
+    });
+
+    test("can update parameter value", async ({ page }) => {
+        // Check that we can see the graph paths change when a parameter changes and model is re-run
+        const graphPathSelector = ":nth-match(.plotly svg .scatterlayer .scatter g.lines path, 1)";
+        const previousPath = await page.getAttribute(graphPathSelector, "d");
+        expect(previousPath!.startsWith("M0")).toBe(true);
+
+        await page.fill(":nth-match(#model-params input, 1)", "3");
+
+        await expect(await page.locator(".run-tab .run-update-msg")).toHaveText(
+            "Model code has been recompiled or parameters have been updated. Run Model to view updated graph.", {
+                timeout
+            }
+        );
+
+        // Re-run model
+        await page.click("#run-btn");
+        await expect(await page.locator(".run-tab .run-update-msg")).toHaveText("");
+
+        const newPath = await page.getAttribute(graphPathSelector, "d");
+        expect(newPath!.startsWith("M0")).toBe(true);
+        expect(newPath).not.toEqual(previousPath);
+    });
+
+    test("parameters update when compile with new code", async ({ page }) => {
+        // Navigate to code tab
+        await page.click(":nth-match(.wodin-left .nav-tabs a, 1)");
+
+        // Provide new code
+        await page.press(".monaco-editor textarea", "Control+A");
+        await page.press(".monaco-editor textarea", "Delete");
+        await page.fill(".monaco-editor textarea", newValidCode);
+
+        await expect(page.locator(".run-tab .run-update-msg")).toHaveText(
+            "Model code has been updated. Compile code and Run Model to view updated graph.", {
+                timeout
+            }
+        );
+
+        // Compile code
+        await page.click("#compile-btn");
+        await expect(page.locator(".run-tab .run-update-msg")).toHaveText(
+            "Model code has been recompiled or parameters have been updated. Run Model to view updated graph.", {
+                timeout
+            }
+        );
+
+        // Navigate to options tab
+        await page.click(":nth-match(.wodin-left .nav-tabs a, 2)");
+
+        await expect(await page.innerText(":nth-match(#model-params label, 1)")).toBe("b");
```

Comment:
```suggestion
        await expect(page.innerText(":nth-match(#model-params label, 1)")).toBe("b");
```

Reply:
Would the test fail without the inner await ?

Reply:
Hm, no, still seem to need these. It's not that they fail without both awaits (definitely need the inner one), it's that if they do (and should) fail, the failure appears in the report listed against the wrong test.